src/distributor.ts

```diff
@@ -72,9 +73,23 @@ const addSigListeners = (): void => {
       // Refresh the list of subscribers in every worker process
       updateConfigAndSubscriberList()
     }
+    await dbstore.closeDatabase()
+    process.exit(0)
```

Comment:
Actually, we don't exit when SIGUSR1 event is triggered. it's to refresh the config changes